src/Sylius/Behat/Context/Ui/Admin/ManagingPaymentsContext.php

```diff
@@ -0,0 +1,164 @@
+    }
+
+    /**
+     * @When I complete the payment of order :orderNumber
+     */
+    public function iCompleteThePaymentOfOrder(string $orderNumber): void
+    {
+        $this->indexPage->completePaymentOfOrderWithNumber($orderNumber);
+    }
+
+    /**
+     * @When I filter
+     */
+    public function iFilter(): void
+    {
+        $this->indexPage->filter();
+    }
+
+    /**
+     * @Then I should see :count payments in the list
+     */
+    public function iShouldSeePaymentsInTheList(int $count): void
+    {
+        Assert::same($count, $this->indexPage->countItems());
+    }
+
+    /**
+     * @Then the payments of the :orderNumber order should be :paymentState for :customer
+     */
+    public function thePaymentsOfTheOrderShouldBeFor(
+        string $orderNumber,
+        string $paymentState,
+        CustomerInterface $customer,
+        Channel $channel = null
+    ): void {
+        $parameters = [
+            'number' => $orderNumber,
+            'state' => $paymentState,
+            'customer' => $customer->getEmail(),
+        ];
+
+        if ($channel !== null) {
+            $parameters = ['channel' => $channel->getCode()];
+        }
+
+        Assert::true($this->indexPage->isSingleResourceOnPage($parameters));
+    }
+
+    /**
+     * @Then I should see a single payment in the list
+     */
+    public function iShouldSeeASinglePaymentInTheList(): void
+    {
+        Assert::same(1, $this->indexPage->countItems());
+    }
+
+    /**
+     * @Then I should see order page with details of order :order
+     */
+    public function iShouldSeeOrderPageWithDetailsOfOrder(OrderInterface $order): void
+    {
+        Assert::true($this->orderShowPage->isOpen(['id' => $order->getId()]));
+    }
+
+    /**
+     * @Then I should see the payment of the :orderNumber order
+     */
+    public function iShouldSeeThePaymentOfTheOrder(string $orderNumber): void
+    {
+        Assert::true($this->indexPage->isSingleResourceOnPage(['number' => $orderNumber]));
+    }
+
+    /**
+     * @Then I should see the payment of order :orderNumber as :paymentState
+     */
+    public function iShouldSeeThePaymentOfOrderAs(string $orderNumber, string $paymentState): void
+    {
+        Assert::same($paymentState, $this->indexPage->getPaymentStatusByOrderNumber($orderNumber));
+    }
+
+    /**
+     * @Then I should be notified that the payment has been completed
+     */
+    public function iShouldBeNotifiedThatThePaymentHasBeenCompleted(): void
+    {
+        $this->notificationChecker->checkNotification('Payment has been completed.', NotificationType::success());
+    }
+
+    /**
+     * @Then I should not see a payment of order :orderNumber
+     */
+    public function iShouldNotSeeAPaymentOfOrder(string $orderNumber): void
+    {
+        Assert::false($this->indexPage->isSingleResourceOnPage(['number' => $orderNumber]));
+    }
+
+    /**
+     * @Then I move to the details of first payment's order
```

Comment:
```suggestion
     * @When I move to the details of first payment's order
```
And you should move this method before Then steps